packages/synapse-react-client/src/components/AclEditor/useSortResourceAccessList.ts

```diff
@@ -22,7 +22,9 @@ export default function useSortResourceAccessList(
     data: userGroupHeadersOnResourceAccessList,
     isLoading,
     error,
-  } = useGetUserGroupHeaders(principalIdsOnResourceAccessList.map(String))
+  } = useGetUserGroupHeaders(principalIdsOnResourceAccessList.map(String), {
+    enabled: principalIdsOnResourceAccessList.length > 0,
```

Comment:
prevent attempting to fetch usergroupheaders in the case where the parent component passes an empty resource access list